### PR TITLE
*: use xtask workflow

### PR DESCRIPTION
**Pull request by BusyJay**
This PR removes all scripts for maintaining the project. Instead,
introduce [xtask][1] workflow to whole process more portable and rusty.

It also introduces a new workflow named codegen to generate code for
health crate.

[1]: https://github.com/matklad/cargo-xtask

### CODE REVIEW

File: xtask/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    cmd
+}
+
+fn submodule() {
+    exec(cmd("git").args(&["submodule", "update", "--init", "grpc-sys/grpc"]));
+    for dir in &["cares/cares", "abseil-cpp", "re2"] {
+        exec(cmd_in("git", "grpc-sys/grpc/third_party").args(&[
+            "submodule",
+            "update",
+            "--init",
```
Comment (hicqu) [inline xtask/src/main.rs:114]: Should add `--recursive`? I can't compile without the flag if I haven't done it.

**Comment by hicqu** [comment]
It's necessary to make it can be compiled with rustc nightly 2021-01-16, or update toolchain for TiKV.
Rest LGTM!

**Review by hicqu — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Comment by BusyJay** [comment]
xtask is only used for development, so it doesn't matter even if rustc used in TiKV can't compile it.

File: xtask/src/main.rs (hunk 1 above)
Comment (BusyJay) [reply xtask/src/main.rs:114]: It's a for loop executed in the directory, so `recursive` is not necessary. On the other hand, recursive will also clone other unnecessary dependencies.

**Review by hunterlxt — COMMENTED** [review]
LGTM

**Pull request merged**